lib/models/user.rb

```diff
@@ -0,0 +1,34 @@
+require 'bcrypt'
+
+class User
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  include DataMapper::Resource
+
+  has n, :space, through: Resource
+  property :id, Serial
+  property :user_name, String, required: true
+  property :email, String, required: true, unique: true
+  property :password_digest, Text
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(email, password)
+      user = first(email: email)
+    if user && BCrypt::Password.new(user.password_digest) == password
+      user
+    else
+      nil
+    end
```

Comment:
Avoid explicitly returning `nil`. It's a bit of a code smell - why are you passing chunks of nothing around in your program?